espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/w3c/models/ActionSequence.java

```diff
@@ -0,0 +1,51 @@
+package io.appium.espressoserver.lib.helpers.w3c.models;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import io.appium.espressoserver.lib.handlers.exceptions.InvalidArgumentException;
+import io.appium.espressoserver.lib.handlers.exceptions.NotYetImplementedException;
+
+import static io.appium.espressoserver.lib.helpers.w3c.models.W3CActions.processSourceActionSequence;
+
+/**
+ * Pull out the actions from the input sources and group them by 'ticks'
+ *
+ * Defined in 17.3 of spec 'extract an action sequence'
+ */
+public class ActionSequence implements Iterator<Tick> {
+
+    private List<Tick> actionsByTick;
+    private int tickCounter;
+
+    public ActionSequence(W3CActions w3CActions) throws InvalidArgumentException, NotYetImplementedException {
+        tickCounter = 0;
+        actionsByTick = new ArrayList<>();
+
+        // Transpose the actionsByTick within the input sources
+        if (w3CActions.getActions() != null) {
+            for (InputSource inputSource : w3CActions.getActions()) {
+                int i = 0;
```

Comment:
it's not very clear what `i` is

Reply:
I'll change it to `tickIndex`... I have a bad habit of always using `i` for basic iterators.